huaweicloud/services/eg/resource_huaweicloud_eg_connection.go

```diff
@@ -0,0 +1,386 @@
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceConnection() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceConnectionCreate,
+		UpdateContext: resourceConnectionUpdate,
+		ReadContext:   resourceConnectionRead,
+		DeleteContext: resourceConnectionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the name of the connection.`,
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the VPC to which the connection belongs.`,
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the subnet to which the connection belongs.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the description of the connection.`,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Specifies the type of the connection.`,
+			},
+			"kafka_detail": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        connectionKafkaDetailSchema(),
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Specifies the configuration details of the kafka intance.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the status of the connection.`,
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The creation time of the connection.`,
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The last update time of the connection.`,
+			},
+		},
+	}
+}
+
+func connectionKafkaDetailSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the kafka intance.`,
+			},
+			"connect_address": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the IP address of the kafka intance.`,
+			},
+			"sasl_ssl": {
```

Comment:
To be consistent with the Kafka instance resource, we can set this API field automatically by username and password.

Reply:
sure